core/vm/gas_table.go

```diff
@@ -440,7 +440,7 @@ func gasCallCode(evm *EVM, contract *Contract, stack *Stack, mem *Memory, memory
 		address := common.Address(stack.Back(1).Bytes20())
 		transfersValue := !stack.Back(2).IsZero()
 		if transfersValue {
-			gas, overflow = math.SafeAdd(gas, evm.AccessEvents.ValueTransferGas(contract.Address().Bytes()[:], address.Bytes()[:]))
+			gas, overflow = math.SafeAdd(gas, evm.StateDB.AccessEvents().ValueTransferGas(contract.Address().Bytes()[:], address.Bytes()[:]))
```

Comment:
While we're at it, it might make sense to just define a `AccessEvents()` method on interpreter, so that we reduce the verbosity.

Reply:
My hunch is that we might finally define a list of AccessEvent functions in `vm.StateDB` interface, e.g. AddAccountBalanceForRead, AddAccountBalanceForWrite, etc. Probably we finally need to handle the AccessEvent mutation revert? It makes more sense to interact with StateDB imo